source/common/http/filter_manager.cc

```diff
@@ -1335,7 +1337,9 @@ void FilterManager::setBufferLimit(uint32_t new_limit) {
 }
 
 void FilterManager::contextOnContinue(ScopeTrackedObjectStack& tracked_object_stack) {
-  tracked_object_stack.add(connection_);
+  if (connection_.has_value()) {
```

Comment:
this appears to be a semantic change. Were we previously adding potentially null-pointers to the stack?

Reply:
are you asking if I'm fixing a crashing bug in public?  No :-P  the interface allowed for null pointers, but the implementation with a hard-coded Network::Connection& did not.  For upstream filters, there may actually not be a downstream connection in the case of async requests, so for that we have to handle this case.